File: app/src/main/java/com/luisenricke/simpleroomapp/database/dao/UserDAO.java
package com.luisenricke.simpleroomapp.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.luisenricke.simpleroomapp.database.entity.User;

import java.util.List;

@Dao
public abstract class UserDAO {

    @Insert
    public abstract long insert(User row);

    @Insert
    public abstract void inserts(User... row);

    @Insert
    public abstract void inserts(List<User> rows);

    @Update
    public abstract int update(User row);

    @Update
    public abstract int updates(User... rows);

    @Update
    public abstract int updates(List<User> rows);

    @Delete
    public abstract void delete(User row);

    @Delete
    public abstract void deletes(User... row);

    @Delete
    public abstract void deletes(List<User> rows);

    @Query("DELETE FROM User WHERE id = :id")
    public abstract int deleteById(int id);

    @Query("DELETE FROM User")
    public abstract void drop();

    @Query("SELECT * FROM User")
    public abstract List<User> get();

    @Query("SELECT * FROM User WHERE id = :id")
    public abstract User getById(int id);

    @Query("SELECT COUNT(*) FROM User")
    public abstract long count();
}

File: app/src/main/java/com/luisenricke/simpleroomapp/data/dao/UserDAO.kt
package com.luisenricke.simpleroomapp.data.dao

import androidx.room.Dao
import androidx.room.Query
import com.luisenricke.room.dao.Base
import com.luisenricke.room.dao.Delete
import com.luisenricke.room.dao.PrimaryKey
import com.luisenricke.room.dao.Update
import com.luisenricke.simpleroomapp.data.entity.User
import com.luisenricke.simpleroomapp.data.entity.User.SCHEMA

@Suppress("unused")
@Dao
abstract class UserDAO : Base<User>, Update<User>, Delete<User>, PrimaryKey<User> {

    @Query("SELECT COUNT(*) FROM ${SCHEMA.TABLE}")
    abstract override fun count(): Long

    @Query("SELECT * FROM ${SCHEMA.TABLE}")
    abstract override fun get(): List<User>

    @Query("DELETE FROM ${SCHEMA.TABLE}")
    abstract override fun drop()

    @Query("SELECT * FROM ${SCHEMA.TABLE} WHERE id = :id")
    abstract override fun get(id: Long): User

    @Query("SELECT * FROM ${SCHEMA.TABLE} WHERE id IN(:ids)")
    abstract override fun get(ids: LongArray): List<User>

    @Query("DELETE FROM ${SCHEMA.TABLE} WHERE id = :id")
    abstract override fun delete(id: Long): Int

    @Query("DELETE FROM ${SCHEMA.TABLE} WHERE id IN(:ids)")
    abstract override fun deletes(ids: LongArray): Int
}
